packages/analytics/client/src/analytics_client/analytics_client.ts

```diff
@@ -134,7 +134,10 @@ export class AnalyticsClient implements IAnalyticsClient {
     };
 
     // debug-logging before checking the opt-in status to help during development
-    if (this.initContext.isDev) {
+    if (
+      this.initContext.isDev ||
+      process.env.BUILDKITE_PIPELINE_SLUG === 'kibana-single-user-performance-debug'
```

Comment:
This client also works on the browser. Accessing the field straight away may lead to errors.

```suggestion
      process?.env?.BUILDKITE_PIPELINE_SLUG === 'kibana-single-user-performance-debug'
```